invenio_jobs/services/config.py

```diff
@@ -8,66 +8,95 @@
 
-from ..models import Job, Run
+from ..models import Job, Run, Task
+from . import results
 from .links import JobLink
 from .permissions import JobPermissionPolicy, RunPermissionPolicy, TasksPermissionPolicy
-from .schema import JobSchema
+from .schema import JobSchema, TaskSchema
+
+
+class TasksSearchOptions(SearchOptionsBase):
+    """Tasks search options."""
+
+    sort_default = "name"
+    sort_direction_default = "asc"
+    sort_direction_options = {
+        "asc": dict(
+            title=_("Ascending"),
+            fn=partial(sorted, key=lambda t: t.name),
+        ),
+        "desc": dict(
+            title=_("Descending"),
+            fn=partial(sorted, key=lambda t: t.name, reverse=True),
+        ),
+    }
+    sort_options = {"name": dict(title=_("Name"), fields=["name"])}
+
+    pagination_options = {"default_results_per_page": 25}
 
 
 class TasksServiceConfig(ServiceConfig, ConfiguratorMixin):
     """TaskService factory configuration."""
 
-    # Common configuration
     service_id = "tasks"
-    permission_policy_cls = TasksPermissionPolicy
 
-    result_list_cls = RecordList
+    record_cls = Task
+    search = TasksSearchOptions
+    schema = TaskSchema
 
-    links_item = {
-        "self": JobLink("{+api}/tasks"),
-    }
+    permission_policy_cls = FromConfig(
+        "JOBS_TASKS_PERMISSION_POLICY",
+        default=TasksPermissionPolicy,
+    )
+
+    result_list_cls = results.List
+
+    links_item = None
+    links_search = pagination_links("{+api}/tasks{?args*}")
 
 
 class JobSearchOptions(SearchOptionsBase):
     """Job search options."""
 
-    # TODO: See what we need to override
+    sort_default = "title"
+    sort_direction_default = "asc"
+    sort_direction_options = {
+        "asc": dict(title=_("Ascending"), fn=asc),
+        "desc": dict(title=_("Descending"), fn=desc),
+    }
+    sort_options = {"title": dict(title=_("Title"), fields=["title"])}
+
+    pagination_options = {"default_results_per_page": 25}
 
 
 class JobsServiceConfig(ServiceConfig, ConfiguratorMixin):
     """Service factory configuration."""
 
-    # Common configuration
     service_id = "jobs"
-    permission_policy_cls = JobPermissionPolicy
 
-    # TODO: See if we need to define custom Job result item and list classes
-    result_item_cls = RecordItem
-    result_list_cls = RecordList
-
-    # Record specific configuration
     record_cls = Job
-
-    # TODO: See if these are needed since we don't index jobs
-    # indexer_cls = None
-    # indexer_queue_name = None
-    # index_dumper = None
-
-    # Search configuration
     search = JobSearchOptions
-
-    # Service schema
     schema = JobSchema
 
+    permission_policy_cls = FromConfig(
+        "JOBS_PERMISSION_POLICY",
+        default=JobPermissionPolicy,
+    )
+
+    result_item_cls = results.Item
+    result_list_cls = results.List
+
     links_item = {
         "self": JobLink("{+api}/jobs/{id}"),
         "runs": JobLink("{+api}/jobs/{id}/runs"),
```

Comment:
🥺 
```suggestion
        "self_html": JobLink("jobs/{id}"),
        "runs_html": JobLink("jobs/{id}/runs"),
```

Reply:
or should i make the URL myself with the id?

Reply:
makes sense, I was assuming this whole endpoint is admin facing so admins would be the users in this case